builder/amazon/common/ami_config.go

```diff
@@ -8,23 +8,23 @@ import (
 
 // AMIConfig is for common configuration related to creating AMIs.
 type AMIConfig struct {
-	AMIName                 string            `mapstructure:"ami_name"`
-	AMIDescription          string            `mapstructure:"ami_description"`
-	AMIVirtType             string            `mapstructure:"ami_virtualization_type"`
-	AMIUsers                []string          `mapstructure:"ami_users"`
-	AMIGroups               []string          `mapstructure:"ami_groups"`
-	AMIProductCodes         []string          `mapstructure:"ami_product_codes"`
-	AMIRegions              []string          `mapstructure:"ami_regions"`
-	AMISkipRegionValidation bool              `mapstructure:"skip_region_validation"`
-	AMITags                 map[string]string `mapstructure:"tags"`
-	AMIEnhancedNetworking   bool              `mapstructure:"enhanced_networking"`
```

Comment:
I agree, that's exactly what I was thinking.

Maybe spit out some type of deprecation message if the user uses `enhanced_networking` (bool)?

Will also need to deal with the case of someone doing:

```
{
  "enhanced_networking": true,
  "enhanced_networking_type": "ena"
}
```